### ISSUE TITLE
RFE: Reading existing values of "cgroup.*" 

### ISSUE THREAD

**Issue by justinfx** [libcgroup/libcgroup#297]
Hi there. I've been a user of libcgroup with cgroup v1 since 2013, and I am just now updating my custom python bindings to support the new v2 support. But I am struggling to understand the intended way to access "cgroup.*" values that are not specific to the traditional controllers.

This question is very much related to #131 where a similar question was asked regarding setting "cgroup.type". I had worked out from reading the source code that it is valid to add a "cgroup" controller and then I was able to set the value of "cgroup.freezer" to 1. However I am failing to understand how to initially read the values. It seems the "add_value" api wants to modify the value in addition to loading it into the internal structure. And it doesnt seem one can read a value unless its listed in the internal structure for the controller value names. But when I add the "cgroup" controller, the value count is 0.

So then, how does one read a value like "cgroup.freezer" or other "cgroup.*" values before they are written to?

**Comment by drakenclimber** [comment]
Thanks for the question, @justinfx.  We have slowly added support for reading/writing `cgroup.*` files, but as you've noted, it's still lacking in some places.

For example, we really need to support a program like this:
```
#!/usr/bin/env python3
from libcgroup import Cgroup, Version

cg = Cgroup('user.slice', Version.CGROUP_V2)
cg.add_controller('cgroup')
cg.get()

print(cg)
```

But it's not there yet.  The above code will return all "real" controllers and their settings, but the `cgroup.*` files are not returned.  (Note that the exact syntax may change slightly, as I'm not sure if creating a controller called `cgroup` is the best way to go about things.)

I'll add this to our todo list for the v3.0.1 milestone.  Looks like `cgroup_get_cgroup()` needs to be made smarter to handle a cgroup named `cgroup`.  (That's a lot of cgroups :) .)  And as always, patches are more than welcome :)

**Comment by drakenclimber** [comment]
Resolved in pull request #300
